### ISSUE TITLE
Feature Request: Markmap root defined by Frontmatter YAML value 'title:' 

### ISSUE THREAD

**Issue by janklostermann** [markmap/markmap-vscode#52]
Hello,
Thanks so much for your plugin. It is so helpful!

To become more compatible with mindmap-exports to markdown (e.g. from freeplane) and save a level of headline hierarchy I have the following suggestion:

### Situation now
With more than one h1-header (`#`) there is no root of the mindmap. 
export from freeplane:
```markdown
test title
# header 1
## header 1.1
## header 1.2
# header 2
## header 2.2
```
![image](https://github.com/markmap/markmap-vscode/assets/11610051/e46b4618-f02b-4c9f-afea-265e5fc6afbb)

### Suggested improvement
The root of the mindmap could be easily added without refactoring the complete hierarchy by simply using a "title" value from markdowns YAML frontmatter.  With that, I would only need to extend the first line to become the frontmatter.

```markdown
---
title: test title # used as root of the mindmap
---
# header 1
## header 1.1
## header 1.2
# header 2
## header 2.2
```

![image](https://github.com/markmap/markmap-vscode/assets/11610051/08645c6b-3987-433f-b58a-c30944ba2039)

**Comment by janklostermann** [comment]
In case there is only one h1-headline (`#`) in the markdown, it is probably best to ignore the YAML frontmatter value `title` and use this single h1-header as root (as is the current behaviour).

**Comment by gera2ld** [comment]
Good idea.

**Comment by janklostermann** [comment]
Great! 
Thanks so much for implementing this improvement so quickly!

**Comment by gera2ld** [comment]
It will be released in the next version.

**Comment by gera2ld** [comment]
Fixed in v0.1.4

**Comment by janklostermann** [comment]
Thanks so much! Works like a charm!

Am 21. April 2024 18:02:51 MESZ schrieb Gerald ***@***.***>:
>Fixed in v0.1.4
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/markmap/markmap-vscode/issues/52#issuecomment-2068108352
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- Von /e/ Mail gesendet.
